src/metric-engine/src/engine/alter/extract_new_columns.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{HashMap, HashSet};
+
+use store_api::metadata::ColumnMetadata;
+use store_api::region_request::{AlterKind, RegionAlterRequest};
+use store_api::storage::{ColumnId, RegionId};
+
+use crate::error::Result;
+
+/// Extract new columns from the create requests.
+///
+/// # Panics
+///
+/// This function will panic if the alter kind is not `AddColumns`.
+pub fn extract_new_columns<'a>(
```

Comment:
What about using a hashemap instead of hashset and vec, then just let the hashmap as the return value and remove the new_column_names new_columns args?